hera_cal/redcal.py

```diff
@@ -460,10 +460,11 @@ def remove_degen(self, antpos, sol, degen_sol=None):
 
         #Amplitude renormalization: fixes the mean abs product of gains (as they appear in visibilities)
         for antpol in antpols:
+            bls_for_average = [bl for bls in self.reds for bl in bls if (bl[2] == 2*antpol) and (bls[0] in bl_pairs)]
             meanSqAmplitude = np.mean([np.abs(g[(ant1,pol[0])] * g[(ant2,pol[1])])
-                for (ant1,ant2,pol) in bl_pairs if pol == 2*antpol], axis=0)
+                for (ant1,ant2,pol) in bls_for_average], axis=0)
             degenMeanSqAmplitude = np.mean([np.abs(degen_sol[(ant1,pol[0])] * degen_sol[(ant2,pol[1])])
-                for (ant1,ant2,pol) in bl_pairs if pol == 2*antpol], axis=0)
+                for (ant1,ant2,pol) in bls_for_average], axis=0)
             gainSols[gainPols == antpol] *= (degenMeanSqAmplitude / meanSqAmplitude)**.5
             visSols[visPols[:,0] == antpol] *= (meanSqAmplitude / degenMeanSqAmplitude)**.5
```

Comment:
Should the visSols be multiplied by the ratio^1 instead of ratio^.5

Reply:
It's ratio**.5, but it's done twice...once first pol and once for the second. So `visSols[visPols[:,0] == 'xx'` for example will select all visibilities with 'xx' in the first position.  This allows us to remove degen from xy and yx visibilities too.

Reply:
Got it